packages/clerk-js/src/core/clerk.ts

```diff
@@ -1324,125 +1359,14 @@ export class Clerk implements ClerkInterface {
-
-    // Handles multi-session scenario when switching between `pending` sessions
-    // and satisfying task requirements such as organization selection
-    if (inActiveBrowserTab() || !this.#options.standardBrowser) {
-      await this.#touchCurrentSession(session);
-      newSession = this.#getSessionFromClient(session.id) ?? session;
-    }
-
-    // Syncs __session and __client_uat, in case the `pending` session
-    // has expired, it needs to trigger a sign-out
-    const token = await session.getToken();
-    if (!token) {
-      eventBus.emit(events.TokenUpdate, { token: null });
-    }
-
-    // Only triggers navigation for internal AIO components routing or custom URLs
-    const shouldNavigateOnSetActive = this.#componentNavigationContext;
-    if (newSession?.currentTask && shouldNavigateOnSetActive) {
-      await navigateToTask(session.currentTask.key, {
-        options: this.#options,
-        environment: this.environment,
-        globalNavigate: this.navigate,
-        componentNavigationContext: this.#componentNavigationContext,
-      });
-    }
-
-    this.#setAccessors(session);
-    this.#emit();
-
-    await onAfterSetActive();
-  };
-
-  public __internal_navigateToTaskIfAvailable = async ({
-    redirectUrlComplete,
-  }: __internal_NavigateToTaskIfAvailableParams = {}): Promise<void> => {
-    const onBeforeSetActive: SetActiveHook =
-      typeof window !== 'undefined' && typeof window.__unstable__onBeforeSetActive === 'function'
-        ? window.__unstable__onBeforeSetActive
-        : noop;
-
-    const onAfterSetActive: SetActiveHook =
-      typeof window !== 'undefined' && typeof window.__unstable__onAfterSetActive === 'function'
-        ? window.__unstable__onAfterSetActive
-        : noop;
-
-    const session = this.session;
-    if (!session || !this.environment) {
-      return;
-    }
-
-    if (session.status === 'pending') {
-      await navigateToTask(session.currentTask.key, {
-        options: this.#options,
-        environment: this.environment,
-        globalNavigate: this.navigate,
-        componentNavigationContext: this.#componentNavigationContext,
-      });
-      return;
-    }
-
-    await onBeforeSetActive();
-
-    if (redirectUrlComplete) {
-      const tracker = createBeforeUnloadTracker(this.#options.standardBrowser);
-
-      await tracker.track(async () => {
-        if (!this.client) {
-          return;
-        }
-
-        if (this.client.isEligibleForTouch()) {
-          const absoluteRedirectUrl = new URL(redirectUrlComplete, window.location.href);
-          await this.navigate(this.buildUrlWithAuth(this.client.buildTouchUrl({ redirectUrl: absoluteRedirectUrl })));
-        } else {
-          await this.navigate(redirectUrlComplete);
-        }
-      });
-
-      if (tracker.isUnloading()) {
-        return;
+      // Do not revalidate server cache for pending sessions to avoid unmount of `SignIn/SignUp` AIOs when navigating to task
+      if (!sessionIsPending) {
```

Comment:
@nikosdouvlis Here I think we should run `onAfterSetActive` if the `newSession.status === 'active'`

For `setActive({ organization })`, it runs a `touch` and then transitions the session to `active`.

I've just reproduced a case where the control components from the server boundary weren't refreshed with the newest session state.